diff -up evolution-data-server-3.28.5/src/camel/providers/imapx/camel-imapx-server.c.imapx-icloud-mail evolution-data-server-3.28.5/src/camel/providers/imapx/camel-imapx-server.c
--- evolution-data-server-3.28.5/src/camel/providers/imapx/camel-imapx-server.c.imapx-icloud-mail	2021-04-23 09:30:07.734875376 +0200
+++ evolution-data-server-3.28.5/src/camel/providers/imapx/camel-imapx-server.c	2021-04-23 09:31:45.321718451 +0200
@@ -629,6 +629,12 @@ imapx_server_stash_command_arguments (Ca
 		if (CAMEL_IMAPX_HAVE_CAPABILITY (is->priv->cinfo, SPECIAL_USE) || CAMEL_IMAPX_HAVE_CAPABILITY (is->priv->cinfo, X_GM_EXT_1))
 			g_string_append_printf (buffer, " SPECIAL-USE");
 		is->priv->list_return_opts = g_string_free (buffer, FALSE);
+	} else if (!is->priv->is_broken_cyrus && CAMEL_IMAPX_HAVE_CAPABILITY (is->priv->cinfo, LIST_STATUS)) {
+		buffer = g_string_new ("");
+		g_string_append_printf (
+			buffer, "STATUS (%s)",
+			is->priv->status_data_items);
+		is->priv->list_return_opts = g_string_free (buffer, FALSE);
 	} else {
 		is->priv->list_return_opts = NULL;
 	}
@@ -6208,7 +6214,7 @@ camel_imapx_server_list_sync (CamelIMAPX
 
 	camel_imapx_command_unref (ic);
 
-	if (success && !is->priv->list_return_opts) {
+	if (success && (!is->priv->list_return_opts || CAMEL_IMAPX_LACK_CAPABILITY (is->priv->cinfo, LIST_EXTENDED))) {
 		ic = camel_imapx_command_new (is, CAMEL_IMAPX_JOB_LSUB, "LSUB \"\" %s",
 			pattern);
 
diff -up evolution-data-server-3.28.5/src/camel/providers/imapx/camel-imapx-utils.c.imapx-icloud-mail evolution-data-server-3.28.5/src/camel/providers/imapx/camel-imapx-utils.c
--- evolution-data-server-3.28.5/src/camel/providers/imapx/camel-imapx-utils.c.imapx-icloud-mail	2018-07-30 15:17:06.000000000 +0200
+++ evolution-data-server-3.28.5/src/camel/providers/imapx/camel-imapx-utils.c	2021-04-23 09:30:07.736875373 +0200
@@ -580,13 +580,6 @@ imapx_parse_capability (CamelIMAPXInputS
 			stream, &token, &len, cancellable, &local_error);
 	}
 
-	/* Some capabilities are extensions of other capabilities.
-	 * Make sure all prerequisite capability flags are present. */
-
-	/* LIST-STATUS is an extension of LIST-EXTENDED. */
-	if (CAMEL_IMAPX_HAVE_CAPABILITY (cinfo, LIST_STATUS))
-		cinfo->capa |= imapx_lookup_capability ("LIST-EXTENDED");
-
 	if (local_error != NULL) {
 		g_propagate_error (error, local_error);
 		imapx_free_capability (cinfo);
