diff -up evolution-data-server-3.28.5/src/calendar/backends/file/e-cal-backend-file.c.calbackendfile-interval-tree-destroy evolution-data-server-3.28.5/src/calendar/backends/file/e-cal-backend-file.c
--- evolution-data-server-3.28.5/src/calendar/backends/file/e-cal-backend-file.c.calbackendfile-interval-tree-destroy	2021-06-16 16:09:19.863360296 +0200
+++ evolution-data-server-3.28.5/src/calendar/backends/file/e-cal-backend-file.c	2021-06-16 16:10:14.225398710 +0200
@@ -298,7 +298,8 @@ free_calendar_data (ECalBackendFile *cbf
 
 	g_rec_mutex_lock (&priv->idle_save_rmutex);
 
-	e_intervaltree_destroy (priv->interval_tree);
+	if (priv->interval_tree)
+		e_intervaltree_destroy (priv->interval_tree);
 	priv->interval_tree = NULL;
 
 	free_calendar_components (priv->comp_uid_hash, priv->icalcomp);
