diff -up evolution-data-server-3.28.5/src/camel/camel-folder-summary.c.nonstandard-uuencode-encoding evolution-data-server-3.28.5/src/camel/camel-folder-summary.c
--- evolution-data-server-3.28.5/src/camel/camel-folder-summary.c.nonstandard-uuencode-encoding	2018-07-30 15:17:06.000000000 +0200
+++ evolution-data-server-3.28.5/src/camel/camel-folder-summary.c	2021-09-30 17:30:47.674951472 +0200
@@ -3153,7 +3153,8 @@ summary_traverse_content_with_parser (Ca
 					else
 						camel_mime_filter_reset (summary->priv->filter_qp);
 					enc_id = camel_mime_parser_filter_add (mp, summary->priv->filter_qp);
-				} else if (!g_ascii_strcasecmp (encoding, "x-uuencode")) {
+				} else if (!g_ascii_strcasecmp (encoding, "x-uuencode") ||
+					   !g_ascii_strcasecmp (encoding, "uuencode")) {
 					d (printf (" decoding x-uuencode\n"));
 					if (summary->priv->filter_uu == NULL)
 						summary->priv->filter_uu = camel_mime_filter_basic_new (CAMEL_MIME_FILTER_BASIC_UU_DEC);
diff -up evolution-data-server-3.28.5/src/camel/camel-mime-utils.c.nonstandard-uuencode-encoding evolution-data-server-3.28.5/src/camel/camel-mime-utils.c
--- evolution-data-server-3.28.5/src/camel/camel-mime-utils.c.nonstandard-uuencode-encoding	2018-07-30 15:17:06.000000000 +0200
+++ evolution-data-server-3.28.5/src/camel/camel-mime-utils.c	2021-09-30 17:30:47.674951472 +0200
@@ -3268,6 +3268,9 @@ camel_transfer_encoding_from_string (con
 		for (i = 0; i < G_N_ELEMENTS (encodings); i++)
 			if (!g_ascii_strcasecmp (string, encodings[i]))
 				return i;
+
+		if (!g_ascii_strcasecmp (string, "uuencode"))
+			return CAMEL_TRANSFER_ENCODING_UUENCODE;
 	}
 
 	return CAMEL_TRANSFER_ENCODING_DEFAULT;
