diff -up evolution-data-server-3.28.5/src/modules/secret-monitor/module-secret-monitor.c.secret-monitor-warnings evolution-data-server-3.28.5/src/modules/secret-monitor/module-secret-monitor.c
--- evolution-data-server-3.28.5/src/modules/secret-monitor/module-secret-monitor.c.secret-monitor-warnings	2018-07-30 15:17:06.000000000 +0200
+++ evolution-data-server-3.28.5/src/modules/secret-monitor/module-secret-monitor.c	2021-11-22 12:52:08.683986536 +0100
@@ -167,7 +167,7 @@ secret_monitor_scan_secrets_thread (gpoi
 	g_list_free_full (list, (GDestroyNotify) g_object_unref);
 
 	if (local_error != NULL) {
-		g_warning ("%s: %s", G_STRFUNC, local_error->message);
+		e_source_registry_debug_print ("%s: %s", G_STRFUNC, local_error->message);
 		g_error_free (local_error);
 	}
 
@@ -206,7 +206,7 @@ secret_monitor_scan_secrets_timeout_cb (
 		g_thread_unref (thread);
 
 	if (local_error != NULL) {
-		g_warning ("%s: %s", G_STRFUNC, local_error->message);
+		e_source_registry_debug_print ("%s: %s", G_STRFUNC, local_error->message);
 		g_error_free (local_error);
 		g_object_unref (server);
 	}
